src/rust/cryptography-x509/src/common.rs

```diff
@@ -235,6 +248,24 @@ pub struct DHParams<'a> {
     pub g: asn1::BigUint<'a>,
     pub q: Option<asn1::BigUint<'a>>,
 }
+
+// Based on OpenSSL's `DHparams`
+#[derive(asn1::Asn1Read, asn1::Asn1Write, Clone, PartialEq, Eq, Debug, Hash)]
+pub struct BasicDHParams<'a> {
+    pub p: asn1::BigUint<'a>,
+    pub g: asn1::BigUint<'a>,
+    // No idea what this is for/what it means.
```

Comment:
This is the bit length of the private value, see PKCS#3, section 6 and 7.1. The central authority could optionally specify that `x` be selected of this length, in addition to `0 < x < p-1`

Reply:
Oh perfect, finally have a citation for the underlying sequence structure. Perfect.

Reply:
Thank you!